internal/pkg/aws/ec2/ec2.go

```diff
@@ -437,3 +438,25 @@ func (idx *routeTableIndex) IsPublicSubnet(subnetID string) bool {
 	}
 	return idx.mainTable.HasIGW()
 }
+
+// GetManagedPrefixListId returns the PrefixListId of a prefix list queried by name.
+func (c *EC2) GetManagedPrefixListId(prefixListName string) (*string, error) {
+	prefixListOutput, err := c.client.DescribeManagedPrefixLists(&ec2.DescribeManagedPrefixListsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("prefix-list-name"),
+				Values: aws.StringSlice([]string{prefixListName}),
+			},
+		},
+	})
+
+	if err != nil {
+		return nil, fmt.Errorf("query returned error: %w", err)
+	}
+
+	if len(prefixListOutput.PrefixLists) != 1 {
+		return nil, fmt.Errorf("query `%s` did not return one prefix list", prefixListName)
```

Comment:
I would suggest splitting this error into two: 0 and more than 1, for clearer error msg. Like "cannot find any prefix list with the name %s" and "found more than one prefix list with the name %s: %v" and %v is the prefix list IDs we got.